Build failure: pylzma

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

`nix-shell -p python312Packages.pylzma`
or
`nix-shell -p python313Packages.pylzma`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/292383037

### Relevant log output

```console
building '/nix/store/73ijx6mgnsa32rpi0hw009c92c4smgqf-python3.13-pylzma-0.5.0.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Running phase: unpackPhase
unpacking source archive /nix/store/q3gmfw08cb1q0rxcpgjlm7fs24q91spp-pylzma-0.5.0.tar.gz
source root is pylzma-0.5.0
setting SOURCE_DATE_EPOCH to timestamp 1537987412 of file "pylzma-0.5.0/setup.cfg"
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Executing setuptoolsBuildPhase
setup.py build flags: ''
/nix/store/49hg6cgj179pabdiyjy23yz3byw6axi0-python3.13-setuptools-75.8.0/lib/python3.13/site-packages/setuptools/_distutils/dist.py:270: UserWarning: Unknown distribution option: 'tests_require'
  warnings.warn(msg)
/nix/store/49hg6cgj179pabdiyjy23yz3byw6axi0-python3.13-setuptools-75.8.0/lib/python3.13/site-packages/setuptools/_distutils/dist.py:270: UserWarning: Unknown distribution option: 'test_suite'
  warnings.warn(msg)
running bdist_wheel
running build
running build_py
creating build/lib.linux-x86_64-cpython-313
copying py7zlib.py -> build/lib.linux-x86_64-cpython-313
running build_ext
setup.py:105: UnsupportedPlatformWarning: Multithreading is not supported on the platform "linux",
please contact mail@joachim-bauch.de for more informations.
  warn("""\
building 'pylzma' extension
creating build/temp.linux-x86_64-cpython-313/src/compat
creating build/temp.linux-x86_64-cpython-313/src/pylzma
creating build/temp.linux-x86_64-cpython-313/src/sdk/C
gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/nix/store/h8wi6kwl1cywxr3f22836vgrrb11fvlb-libxcrypt-4.4.38/include -fPIC -DPY_SSIZE_T_CLEAN=1 -DWITH_COMPAT=1 -DPYLZMA_VERSION=0.5.0 -D_7ZIP_ST=1 -Isrc/sdk/C -I/nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13 -c src/compat/LzmaCompatDecode.c -o build/temp.linux-x86_64-cpython-313/src/compat/LzmaCompatDecode.o
gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/nix/store/h8wi6kwl1cywxr3f22836vgrrb11fvlb-libxcrypt-4.4.38/include -fPIC -DPY_SSIZE_T_CLEAN=1 -DWITH_COMPAT=1 -DPYLZMA_VERSION=0.5.0 -D_7ZIP_ST=1 -Isrc/sdk/C -I/nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13 -c src/pylzma/pylzma.c -o build/temp.linux-x86_64-cpython-313/src/pylzma/pylzma.o
src/pylzma/pylzma.c: In function ‘PyInit_pylzma’:
src/pylzma/pylzma.c:296:5: warning: ‘PyEval_InitThreads’ is deprecated []
  296 |     PyEval_InitThreads();
      |     ^~~~~~~~~~~~~~~~~~
In file included from /nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13/Python.h:124,
                 from src/pylzma/pylzma.c:26:
/nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13/ceval.h:114:37: note: declared here
  114 | Py_DEPRECATED(3.9) PyAPI_FUNC(void) PyEval_InitThreads(void);
      |                                     ^~~~~~~~~~~~~~~~~~
gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/nix/store/h8wi6kwl1cywxr3f22836vgrrb11fvlb-libxcrypt-4.4.38/include -fPIC -DPY_SSIZE_T_CLEAN=1 -DWITH_COMPAT=1 -DPYLZMA_VERSION=0.5.0 -D_7ZIP_ST=1 -Isrc/sdk/C -I/nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13 -c src/pylzma/pylzma_aes.c -o build/temp.linux-x86_64-cpython-313/src/pylzma/pylzma_aes.o
src/pylzma/pylzma_aes.c:158:5: error: initialization of ‘long int’ from ‘void *’ makes integer from pointer without a cast []
  158 |     NULL,                                /* printfunc  tp_print;   */
      |     ^~~~
src/pylzma/pylzma_aes.c:158:5: note: (near initialization for ‘CAESDecrypt_Type.tp_vectorcall_offset’)
error: command '/nix/store/gj9lra51hwhxnhz05jqk5lh03wipamv0-gcc-wrapper-14-20241116/bin/gcc' failed with exit code 1
error: builder for '/nix/store/73ijx6mgnsa32rpi0hw009c92c4smgqf-python3.13-pylzma-0.5.0.drv' failed with exit code 1;
       last 25 log lines:
       > running build_ext
       > setup.py:105: UnsupportedPlatformWarning: Multithreading is not supported on the platform "linux",
       > please contact mail@joachim-bauch.de for more informations.
       >   warn("""\
       > building 'pylzma' extension
       > creating build/temp.linux-x86_64-cpython-313/src/compat
       > creating build/temp.linux-x86_64-cpython-313/src/pylzma
       > creating build/temp.linux-x86_64-cpython-313/src/sdk/C
       > gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/nix/store/h8wi6kwl1cywxr3f22836vgrrb11fvlb-libxcrypt-4.4.38/include -fPIC -DPY_SSIZE_T_CLEAN=1 -DWITH_COMPAT=1 -DPYLZMA_VERSION=0.5.0 -D_7ZIP_ST=1 -Isrc/sdk/C -I/nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13 -c src/compat/LzmaCompatDecode.c -o build/temp.linux-x86_64-cpython-313/src/compat/LzmaCompatDecode.o
       > gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/nix/store/h8wi6kwl1cywxr3f22836vgrrb11fvlb-libxcrypt-4.4.38/include -fPIC -DPY_SSIZE_T_CLEAN=1 -DWITH_COMPAT=1 -DPYLZMA_VERSION=0.5.0 -D_7ZIP_ST=1 -Isrc/sdk/C -I/nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13 -c src/pylzma/pylzma.c -o build/temp.linux-x86_64-cpython-313/src/pylzma/pylzma.o
       > src/pylzma/pylzma.c: In function ‘PyInit_pylzma’:
       > src/pylzma/pylzma.c:296:5: warning: ‘PyEval_InitThreads’ is deprecated []
       >   296 |     PyEval_InitThreads();
       >       |     ^~~~~~~~~~~~~~~~~~
       > In file included from /nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13/Python.h:124,
       >                  from src/pylzma/pylzma.c:26:
       > /nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13/ceval.h:114:37: note: declared here
       >   114 | Py_DEPRECATED(3.9) PyAPI_FUNC(void) PyEval_InitThreads(void);
       >       |                                     ^~~~~~~~~~~~~~~~~~
       > gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/nix/store/h8wi6kwl1cywxr3f22836vgrrb11fvlb-libxcrypt-4.4.38/include -fPIC -DPY_SSIZE_T_CLEAN=1 -DWITH_COMPAT=1 -DPYLZMA_VERSION=0.5.0 -D_7ZIP_ST=1 -Isrc/sdk/C -I/nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13 -c src/pylzma/pylzma_aes.c -o build/temp.linux-x86_64-cpython-313/src/pylzma/pylzma_aes.o
       > src/pylzma/pylzma_aes.c:158:5: error: initialization of ‘long int’ from ‘void *’ makes integer from pointer without a cast []
       >   158 |     NULL,                                /* printfunc  tp_print;   */
       >       |     ^~~~
       > src/pylzma/pylzma_aes.c:158:5: note: (near initialization for ‘CAESDecrypt_Type.tp_vectorcall_offset’)
       > error: command '/nix/store/gj9lra51hwhxnhz05jqk5lh03wipamv0-gcc-wrapper-14-20241116/bin/gcc' failed with exit code 1
       For full logs, run 'nix log /nix/store/73ijx6mgnsa32rpi0hw009c92c4smgqf-python3.13-pylzma-0.5.0.drv'.
```

### Additional context

Also happens in all python versions in nixpkgs - 3.9 to 3.14


### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.6, NixOS, 25.05 (Warbler), 25.05beta767267.6607cf789e54`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Notify maintainers

@dandellion

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pylzma/default.nix
+++ b/pkgs/development/python-modules/pylzma/default.nix
@@ -2,6 +2,7 @@
   lib,
   buildPythonPackage,
   fetchPypi,
+  fetchpatch,
 }:
 
 buildPythonPackage rec {
@@ -16,6 +17,14 @@ buildPythonPackage rec {
     inherit pname version;
     sha256 = "074anvhyjgsv2iby2ql1ixfvjgmhnvcwjbdz8gk70xzkzcm1fx5q";
   };
+
+  patches = [
+    # https://github.com/fancycode/pylzma/pull/82/
+    (fetchpatch {
+      url = "https://github.com/fancycode/pylzma/commit/2fe0a4ed0588fd572931da4be10ad955636afde4.patch";
+      hash = "sha256-sWdMAmOPVTDnxNTjzPlqQYxqnjmRpK+OqwWF6jpXvIw=";
+    })
+  ];
 
   pythonImportsCheck = [ "pylzma" ];
 

